tools/asdf/src/proto.rs

```diff
@@ -0,0 +1,272 @@
+
+    let version = input.context.version;
+
+    // Set asdf environment variables
+    let install_download_path = &input.context.tool_dir.real_path().unwrap().into_os_string().into_string().unwrap();
+    let cores = if env.os.is_mac() {
+        exec_command!("sysctl -n hw.physicalcpu").stdout
+    } else {
+        exec_command!("nproc").stdout
+    };
+    instructions.append(&mut vec![
+        BuildInstruction::SetEnvVar("ASDF_INSTALL_TYPE".into(), "version".into()),
+        BuildInstruction::SetEnvVar("ASDF_INSTALL_VERSION".into(), version.clone().into()),
+        BuildInstruction::SetEnvVar("ASDF_INSTALL_PATH".into(), install_download_path.clone()),
+        BuildInstruction::SetEnvVar("ASDF_DOWNLOAD_PATH".into(), install_download_path.clone()),
+        BuildInstruction::SetEnvVar("ASDF_CONCURRENCY".into(), cores),
+    ]);
+
+    let id = get_id(None)?;
+
+    // In older versions of asdf there may not be a 'download' script,
+    // instead both download and install were done in the 'install' script.
+    // However, in newer versions, there's two separate 'download' and 'install' scripts.
+    if send_request!(format!("{}/blob/{}/bin/download", repo.url, repo.default_branch)).status == 200 {
+            let download_id = String::from(format!("{id}-{version}-download"));
+            instructions.push(BuildInstruction::InstallBuilder(Box::new(BuilderInstruction {
+                id: download_id.clone(),
+                exe: "bin/download".into(),
+                git: git.clone(),
+                ..BuilderInstruction::default()
+            })));
+            
+            instructions.push(
+                BuildInstruction::RunCommand(Box::new(CommandInstruction::with_builder(
+                    &download_id,
+                    [""],
+                ))),
+            )
+        }
+
+    let install_id = String::from(format!("{id}-{version}-install"));
+    instructions.push(
+        BuildInstruction::InstallBuilder(Box::new(BuilderInstruction {
+            id: install_id.clone(),
+            exe: "bin/install".into(),
+            git,
+            ..BuilderInstruction::default()
+        }))
+    );
+    instructions.push(
+        BuildInstruction::RunCommand(Box::new(CommandInstruction::with_builder(
+            &install_id,
+            [""],
+        ))),
+    );
+
+    let output = BuildInstructionsOutput {
+        instructions,
+        ..Default::default()
+    };
+
+    Ok(Json(output))
+}
+
+#[plugin_fn]
+pub fn locate_executables(
+    Json(input): Json<LocateExecutablesInput>,
+) -> FnResult<Json<LocateExecutablesOutput>> {
+    let exe = get_executable_name()?;
+
+    let install_path = input.context.tool_dir.real_path().unwrap().into_os_string().into_string().unwrap();
+    Ok(Json(LocateExecutablesOutput {
+        exes: HashMap::from_iter([(
+            exe.clone(),
+            ExecutableConfig::new_primary(
+                format!("{install_path}/bin/{exe}")
```

Comment:
These paths should be relative from the install/tool dir, so you don't need to make them absolute like this. `bin/{exe}` should be enough.